client/client_test.go

```diff
@@ -174,12 +174,12 @@ func TestParseHostURL(t *testing.T) {
 }
 
 func TestNewClientWithOpsFromEnvSetsDefaultVersion(t *testing.T) {
-	defer env.PatchAll(t, map[string]string{
+	env.PatchAll(t, map[string]string{
 		"DOCKER_HOST":        "",
 		"DOCKER_API_VERSION": "",
 		"DOCKER_TLS_VERIFY":  "",
 		"DOCKER_CERT_PATH":   "",
-	})()
+	})
```

Comment:
Is `env.PatchAll` not deprecated?  Seems like it'd be good to replace this with `t.Setenv` too.

Reply:
No, that one's not deprecated, and also does slightly more (which could actually be a nice addition to go's standard tooling;

- it resets the environment (`os.ClearEnv()`)
- patches/sets all of the given env-vars
- restores the environment afterwards

Reply:
Was the removal of the `defer` intentional here?  The other calls to `env.PatchAll` in the file had the `defer` and were also changed to `t.Setenv`.

Reply:
Yes; newer versions of gotest.tools use `t.CleanUp()` to reset state automatically (I could've left out that part in this PR  when I extracted it from https://github.com/moby/moby/pull/43340)

Reply:
Got it, thanks for the explanation!